src/main/java/net/yogstation/yogbot/listeners/interactions/slash/BanCommand.kt

```diff
@@ -0,0 +1,27 @@
+package net.yogstation.yogbot.listeners.interactions.slash
+
+import discord4j.common.util.Snowflake
+import discord4j.core.`object`.command.ApplicationCommandInteractionOption
+import discord4j.core.`object`.command.ApplicationCommandInteractionOptionValue
+import discord4j.core.event.domain.interaction.ChatInputInteractionEvent
+import net.yogstation.yogbot.listeners.interactions.ISlashCommand
+import net.yogstation.yogbot.listeners.interactions.SoftbanCommand
+import net.yogstation.yogbot.permissions.PermissionsManager
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+
+@Component
+class BanCommand(
+	private val softbanCommand: SoftbanCommand,
+	private val permissions: PermissionsManager
+): ISlashCommand {
+	override val name = "ban"
+
+	override fun handle(event: ChatInputInteractionEvent): Mono<*> {
+		val userid: Snowflake = event.getOption("person").flatMap(ApplicationCommandInteractionOption::getValue)

```

Comment:
Just wondering if it would be a bit clearer if we didn't use `orElse` here and instead make userId an `Optional<Snowflake>` instead of instantiating a snowflake with an id of 0. Then use something like `userId.empty()` to check if a value is submited.

Not too familiar with Kotling streams, if you got with orElse would you be left with a Optional?

Reply:
Userid is an optional, hence the orElse. The option is required, so it should never be missing, but the API still gives it as optional anyway.